Update to micronaut 5

lissam1 opened the pull request:
Closes: #721

lissam1 committed:
update to micronaut 5.0.0-M2

lissam1 committed:
fix method signature

lissam1 committed:
force spock version

lissam1 committed:
update micronaut validation

lissam1 committed:
remove kotlin

lissam1 committed:
use non-nullable parameters

lissam1 committed:
use spock-groovy 5

sdelamo commented on buildSrc/src/main/groovy/io.micronaut.build.internal.jms-module.gradle:
```diff
@@ -13,3 +13,9 @@ dependencies {
+//TODO remove once Micronaut Test ships Spock version compatible with Groovy 5
+configurations.all {
+    resolutionStrategy {
+        force("org.spockframework:spock-core:2.4-M7-groovy-5.0")
```
```suggestion
        force("org.spockframework:spock-core:2.4-groovy-5.0")
```

sdelamo commented on docs-examples/example-groovy/build.gradle:
```diff
@@ -4,9 +4,13 @@ plugins {
 }
 
 micronaut {
-    testRuntime 'spock'
```
why ?

sdelamo replied:
testRuntime 'spock'

sdelamo commented on gradle/libs.versions.toml:
```diff
@@ -2,25 +2,19 @@
 managed-artemis-jakarta-client = '2.42.0'
-micronaut-docs = "3.0.0"
-micronaut = "4.10.9"
+micronaut = "5.0.0-M2"
```
```suggestion
micronaut = "5.0.0-M3"
```

sdelamo commented on docs-examples/example-groovy/build.gradle:
```diff
@@ -4,9 +4,13 @@ plugins {
 }
+
+dependencies {
+    testImplementation("org.spockframework:spock-core:2.4-M7-groovy-5.0")
```
```suggestion
    testImplementation("org.spockframework:spock-core:2.4-groovy-5.0")
```

sdelamo committed:
Update gradle/libs.versions.toml

sdelamo committed:
Update docs-examples/example-groovy/build.gradle

sdelamo committed:
id("io.micronaut.build.shared.settings") version "8.0.0-M13"

sdelamo committed:
Update buildSrc/src/main/groovy/io.micronaut.build.internal.jms-module.gradle

sdelamo committed:
revert

sdelamo committed:
Merge branch '5.0.x' into update-to-micronaut-5

sdelamo committed:
micronaut-test-resources = "3.0.0-M6"

sdelamo committed:
testcontainers-junit-jupiter

sdelamo committed:
M17

lissam1 committed:
try fix jms

lissam1 committed:
Merge remote-tracking branch 'origin/update-to-micronaut-5' into update-to-micronaut-5

# Conflicts:
#	gradle/libs.versions.toml

lissam1 committed:
add properties

lissam1 committed:
clean up

lissam1 committed:
make test pass

sdelamo commented on tests/tasks-activemq-artemis/src/main/resources/application.yml:
```diff
@@ -1,18 +1,18 @@
-micronaut:
```
delete

sdelamo commented on tests/tasks-activemq-classic/src/main/resources/application.yml:
```diff
@@ -3,11 +3,11 @@ micronaut:
-      image-name: rmohr/activemq:5.15.9
-      exposed-ports:
-        - activemq.port: 61616
+#        connection-string: "tcp://localhost:${activemq.port}"
```
delete

sdelamo committed:
Merge branch '5.0.x' into update-to-micronaut-5

sdelamo committed:
Replace with broker-specific @EachBean factories

Replace JMSConnectionFactoryBeanProcessor with broker-specific @EachBean factories and migrate Kotlin tests to
  JUnit 5

  Replace the generic BeanDefinitionProcessor<JMSConnectionFactory> with
  per-broker @EachBean factory classes (JmsConnectionPoolActiveMQConnectionFactory,
  JmsConnectionPoolActiveMQJMSConnectionFactory, JmsConnectionPoolSQSConnectionFactory)
  to avoid StackOverflowError caused by JMSConnectionPool implementing ConnectionFactory.

  Simplify AbstractJMSListenerMethodProcessor by removing the fallback
  pool registration logic now that pools are created via @EachBean.

  Migrate example-kotlin tests from KoTest to JUnit 5 for consistency
  with the Java and Groovy example modules.

  Update documentation to reflect the new broker configuration approach.

sdelamo committed:
uncomment

sdelamo committed:
Merge branch '5.0.x' into update-to-micronaut-5

sdelamo committed:
Merge branch '5.0.x' into update-to-micronaut-5

sdelamo committed:
dep: org.apache.commons:commons-pool2 to v2.13.1

sdelamo committed:
micronaut-logging = "2.0.0-M2"

sdelamo committed:
remove snakeyaml dependency

sdelamo committed:
graalvm reflection for task serialization

sdelamo committed:
diable in native image localstack test
